compiler/dyno/lib/resolution/Resolver.cpp

```diff
@@ -2248,7 +2302,7 @@ bool Resolver::enter(const For* loop) {
         loopResults.push_back(std::move(cur.byPostorder));
       }
 
-      auto paramLoop = new ResolvedParamLoop(loop);
+      auto paramLoop = new ResolvedParamLoop(forLoop);
```

Comment:
Not really part of this PR, but is this ever freed? Should it be?

Reply:
There is definitely a memory-management issue here that warrants further discussion. I think that discussion is a bit complicated and outside of the scope of this PR, so I will merge presently and take the discussion of memory management for ResolvedParamLoop offline.